File: api-core/models/article/article.go
package article

import (
	"api-core/models"
	"api-core/models/category"
	"api-core/models/dto"
	"github.com/astaxie/beego/orm"
	"time"
)

func SelectById(id int) *models.Article {
	entity := models.Article{
		BaseModel: models.BaseModel{
			Id: id,
		},
	}
	o := orm.NewOrm()
	err := o.Read(&entity)
	if err != nil {
		return nil
	}
	o.LoadRelated(&entity, "Category")
	return &entity
}

func ListByCategoryId(categoryId int) []*models.Article {
	var articleList []*models.Article
	o := orm.NewOrm()
	if categoryId > 0 {
		o.QueryTable(new(models.Article)).Filter("category__id", categoryId).OrderBy("createTime").All(&articleList)
	} else {
		o.QueryTable(new(models.Article)).OrderBy("createTime").All(&articleList)
	}

	return articleList
}

func Save(dto *dto.ArticleDTO) bool {
	o := orm.NewOrm()
	id := dto.Id
	cat := category.SelectById(dto.CategoryId)
	if id > 0 {
		entity := SelectById(id)
		if entity != nil {
			entity.UpdateTime = time.Now()
			entity.Title = dto.Title
			entity.Content = dto.Content
			entity.Category = cat
			if _, err := o.Update(entity); err != nil {
				return false
			}
		}
	} else {
		entity := models.Article{
			BaseModel: models.BaseModel{
				CreateTime: time.Now(),
				UpdateTime: time.Now(),
			},
			Title:    dto.Title,
			Content:  dto.Content,
			Category: cat,
		}
		if _, err := o.Insert(&entity); err != nil {
			return false
		}
	}
	return true

}

File: web/src/model/Article.ts
import Category from "@/model/Category";

export default class Article{
    private _id!:number;
    private _title!:string;
    private _content!:string;
    private _createTime!:string;
    private _updateTime!:string;
    private _category!:Category;


    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get title(): string {
        return this._title;
    }

    set title(value: string) {
        this._title = value;
    }

    get content(): string {
        return this._content;
    }

    set content(value: string) {
        this._content = value;
    }

    get createTime(): string {
        return this._createTime;
    }

    set createTime(value: string) {
        this._createTime = value;
    }

    get updateTime(): string {
        return this._updateTime;
    }

    set updateTime(value: string) {
        this._updateTime = value;
    }

    get category(): Category {
        return this._category;
    }

    set category(value: Category) {
        this._category = value;
    }
}